virtcontainers/sandbox.go

```diff
@@ -1619,19 +1622,23 @@ func (s *Sandbox) updateResources() error {
 		},
 	}
 
+	var mCPU uint32
+
+	// Calculate running total of memory and mCPUs requested
 	for _, c := range s.config.Containers {
 		if m := c.Resources.Memory; m != nil && m.Limit != nil {
 			*sumResources.Memory.Limit += *m.Limit
 		}
 		if cpu := c.Resources.CPU; cpu != nil {
 			if cpu.Period != nil && cpu.Quota != nil {
-				*sumResources.CPU.Period += *cpu.Period
-				*sumResources.CPU.Quota += *cpu.Quota
+				mCPU += uint32(utils.CalculatemCPUs(*cpu.Quota, *cpu.Period))
```

Comment:
Please sync data types with `CalculatemCPUs` instad of do casting

Reply:
done